app/src/main/java/protect/card_locker/preferences/SettingsActivity.kt

```diff
@@ -1,212 +1,190 @@
+                        it.value = ""
+                    } else {
+                        // Need to set preference's value to one of localePreference.getEntryValues() to match the locale.
+                        // Locale.toLanguageTag() theoretically should be one of the values in localePreference.getEntryValues()...
+                        // But it doesn't work for some locales. so trying something more heavyweight.
+
+                        // Obtain all locales supported by the app.
+                        val appLocales = entryValues.map { entry -> Utils.stringToLocale(entry.toString()) }
+                        // Get the app locale that best matches the system one
+                        val bestMatchLocale = Utils.getBestMatchLocale(appLocales, sysLocale)
+                        // Get its index in supported locales
+                        val index = appLocales.indexOf(bestMatchLocale)
+                        // Set preference value to entry value at that index
+                        it.value = entryValues[index].toString()
+                    }
                 }
             }
 
-            localePreference.setOnPreferenceChangeListener((preference, newValue) -> {
+            localePreference?.setOnPreferenceChangeListener { _, newValue ->
                 // See corresponding comment in Utils.updateBaseContextLocale for Android 6- notes
                 if (Build.VERSION.SDK_INT < Build.VERSION_CODES.N) {
-                    refreshActivity(true);
-                    return true;
+                    refreshActivity(true)
+                    return@setOnPreferenceChangeListener true
                 }
-                String newLocale = (String) newValue;
+                val newLocale = newValue as String
                 // If newLocale is empty, that means "System" was selected
-                AppCompatDelegate.setApplicationLocales(newLocale.isEmpty() ? LocaleListCompat.getEmptyLocaleList() : LocaleListCompat.create(Utils.stringToLocale(newLocale)));
-                return true;
-            });
+                AppCompatDelegate.setApplicationLocales(if (newLocale.isEmpty()) LocaleListCompat.getEmptyLocaleList() else LocaleListCompat.create(Utils.stringToLocale(newLocale)))
+                true
+            }
 
             // Disable content provider on SDK < 23 since dangerous permissions
             // are granted at install-time
-            Preference contentProviderReadPreference = findPreference(getResources().getString(R.string.settings_key_allow_content_provider_read));
-            assert contentProviderReadPreference != null;
-            contentProviderReadPreference.setVisible(Build.VERSION.SDK_INT >= Build.VERSION_CODES.M);
+            val contentProviderReadPreference = findPreference<Preference>(getString(R.string.settings_key_allow_content_provider_read))
+            contentProviderReadPreference?.isVisible = Build.VERSION.SDK_INT >= Build.VERSION_CODES.M
```

Comment:
I think it's better to use `!!` here because, if this is null, something is really wrong. It is better to crash when things are really wrong than silently misbehave imho, because crashes are easier to find and fix :)